software/minimap2/align/main.nf

```diff
@@ -0,0 +1,56 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process MINIMAP2_ALIGN {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::minimap2=2.17=hed695b0_3" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/minimap2:2.17--hed695b0_3"
+    } else {
+        container "quay.io/biocontainers/minimap2:2.17--hed695b0_3"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+    path reference
+
+    output:
+    tuple val(meta), path("*.paf"), emit: paf
+    path "*.version.txt"          , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    if (meta.single_end) {
+        """
+        minimap2 \\
+            $options.args \\
+            -t $task.cpus \\
+            $reference \\
+            $reads \\
```

Comment:
If that `if/else` statement is just to deal with single-end/paired-end reads I think it would be better to have some logic that deals with that out of the script for simplicity e.g. [here](https://github.com/nf-core/modules/blob/17dd0415f2f57a61c24f6be4ed6e75bf6ad2ec4e/software/salmon/quant/main.nf#L37)

Reply:
Agreed, will put that in the next commit